### PR TITLE
illegal First rank pawns exception!

### PR DESCRIPTION
**Pull request by OfekShochat**
this commit has a test, and also the exception itself. pretty simple, when row is 0 or 8, dont allow pawns.
really the commits should be cleaned up. my master isnt clean..

### CODE REVIEW

File: src/chess/board.cc [hunk 1]
```diff
@@ -1030,6 +1025,8 @@ void ChessBoard::SetFromFen(std::string fen, int* rule50_ply, int* moves) {
       rooks_.set(row, col);
       bishops_.set(row, col);
     } else if (c == 'P' || c == 'p') {
+      if (row == 8 || row == 0)
+        throw std::out_of_range("pawns at the first/eighth rank are forbidden.");
```
Comment (borg323) [inline src/chess/board.cc:1029]: The exception should be in the same form as the others around it.

File: src/chess/board_test.cc [hunk 2]
```diff
@@ -53,6 +51,16 @@ TEST(BoardSquare, BoardSquare) {
   }
 }
 
+TEST(ChessBoard, IllegalFirstRankPawns) {
+  ChessBoard board;
+  try {
+    board.SetFromFen("nqrbkrnr/bnnbnbnn/8/8/8/8/NNNBPNBN/QNRPKPQQ w - - 0 1");
+    FAIL() << "expected pawns at the first/eighth rank are forbidden.";
+  } catch (std::out_of_range const &err) {
+    EXPECT_EQ(err.what(), std::string("pawns at the first/eighth rank are forbidden."));
```
Comment (borg323) [inline src/chess/board_test.cc:60]: Don't check for the exact error, just that it fails.

File: src/chess/board.cc [hunk 3]
```diff
@@ -1,22 +1,17 @@
 /*
   This file is part of Leela Chess Zero.
   Copyright (C) 2018-2019 The LCZero Authors
-
```
Comment (borg323) [inline src/chess/board.cc:4]: Lots of unrelated white space changes.

File: src/chess/board_test.cc (hunk 2 above)
Comment (OfekShochat) [reply src/chess/board_test.cc:60]: How do I do that?

**Review by borg323 — APPROVED** [review]

**Pull request merged**